src/service.rs

```diff
@@ -63,19 +63,20 @@ where
                             &wait_time
                         );
                     }
-                    let (body, content_type) = self.key_extractor.response_error_content(&negative);
-                    let response = actix_web::HttpResponse::TooManyRequests()
-                        .insert_header(content_type)
-                        .insert_header(("x-ratelimit-after", wait_time))
-                        .body(body.clone());
+                    let mut response_bulider = actix_web::HttpResponse::TooManyRequests();
+                    response_bulider.insert_header(("x-ratelimit-after", wait_time));
+                    let response = self
+                        .key_extractor
+                        .exceed_rate_limit_response(&negative, response_bulider);
+
                     future::Either::Left(future::err(
-                        error::InternalError::from_response(body, response).into(),
+                        error::InternalError::from_response("TooManyRequests", response).into(),
```

Comment:
I wonder whether it's better to just return `HttpResponse`. There's no real benefit for using `Error`, the `Error` type probably just adds unnecessary overhead.

I'd update line 23 like this:

```rust
type Future = future::Either<future::Ready<Result<ServiceResponse<B>, HttpResponse>>, S::Future>;
```

Reply:
Does this mean we will remove the `KeyExtractionError`?

https://github.com/AaronErhardt/actix-governor/blob/58bb2a2214eb6ac3932134abd4b600def97c8797/src/key_extractor.rs#L37-L38

Reply:
No, this would only change the error reporting in the `Service` trait implementation. Everything else would be the same.

Reply:
> No, this would only change the error reporting in the `Service` trait implementation. Everything else would be the same.

I think we can't do this

![image](https://user-images.githubusercontent.com/59842932/185795981-452876f9-d617-449c-9631-71cac80df59e.png)

Reply:
You're right... I wonder why they allow you to choose the error type in the first place then, when everything apart from actix_web::Error is unusable...

Let's leave it like it is then.